srcCpp03/perftest_publisher.cxx

```diff
@@ -117,15 +117,55 @@ int perftest_cpp::Run(int argc, char *argv[]) {
 
     if (_PM.get<int>("unbounded") == 0) {
         if (_PM.get<bool>("keyed")) {
-            _MessagingImpl = new RTIDDSImpl<TestDataKeyed_t>();
+            if (_PM.get<bool>("flatdata")) {
+                #ifdef RTI_FLATDATA_AVAILABLE
+                 if (_PM.get<bool>("zerocopy")) {
+                    _MessagingImpl = new RTIDDSImpl_FlatData<TestDataKeyed_ZeroCopy_w_FlatData_t>();
+                 } else {
+                    _MessagingImpl = new RTIDDSImpl_FlatData<TestDataKeyed_FlatData_t>();
+                 }
+                #endif //RTI_FLATDATA_AVAILABLE
```

Comment:
#else, give an error

Reply:
Why do you want to give an error if RTI_FLATDATA_AVAILABLE is not defined?

I have wrapped those lines in a `#ifdef RTI_FLATDATA_AVAILABLE` in case rtiperftest is compiled against a ConnextDDS version previous to 6.0.0. In that case since flatdata is not available, all the related flatdata code wont be compiled.

Does the later make sense?

Reply:
I think what this means is that if you use -flatdata and you don't have flat data available, then, the test will finish silently. It should show a message saying that flat data is not available for that perftest version.

Also, we need to check how this works when using micro.

Reply:
Rather thank giving an error; do not allow using `-flatData` as an option if it is not available:
```C++
    if (_PM.get<bool>("keyed")) {
          #ifdef RTI_FLATDATA_AVAILABLE
            if (_PM.get<bool>("flatdata")) {
                 if (_PM.get<bool>("zerocopy")) {
                    _MessagingImpl = new RTIDDSImpl_FlatData<TestDataKeyed_ZeroCopy_w_FlatData_t>(true);
                 } else {
                    _MessagingImpl = new RTIDDSImpl_FlatData<TestDataKeyed_FlatData_t>();
                 }
            } else {
                _MessagingImpl = new RTIDDSImpl<TestDataKeyed_t>();
            }
          #else 
            _MessagingImpl = new RTIDDSImpl<TestDataKeyed_t>();
          #endif //RTI_FLATDATA_AVAILABLE
        }
```